core/Request.php

```diff
@@ -244,6 +268,60 @@ public function post($index = false, $default = false)
         return $this->post[$index];
     }
 
+    public function delete($index = false, $default = false)
+    {
+
+        if ($index === false && !empty($this->delete)) {
+            return $this->delete;
+        }
+
+        if(isset($this->delete[$index]) && is_array($this->delete[$index])) {
+            return $this->delete[$index];
+        }
+
+        if (!isset($this->delete[$index]) || strlen($this->delete[$index]) <= 0) {
+            return $default;
+        }
+
+        return $this->delete[$index];
+    }
+
+    public function put($index = false, $default = false)
```

Comment:
Are these new methods just copy/paste clones?   All three of these methods (delete,put,server) could be refactored down to just one however the put/delete methods don't really seem to serve a functional purpose at the moment anyway so I don't believe this branch should be merged until it works.